gusto/state.py

```diff
@@ -253,9 +258,12 @@ def dump(self, t=0, pickup=False):
                 self.dumpfile_ll.write(*self.to_dump_latlon)
 
             # compute diagnostics
+            diagnostic_fns = ['min', 'max', 'rms', 'l2']
```

Comment:
Yes, I like this. At the moment we compute all the diagnostics on all of the fields - which I don't mind because it doesn't seem to massively increase runtime and these things would be a pain to compute after running so I'd rather have them than not. However, that could change if we add a lot of diagnostics, some of which might be field specific.